src/Discord.Net/State/StateBehavior.cs

```diff
@@ -0,0 +1,47 @@
+namespace Discord
+{
+    public enum StateBehavior
+    {
+        /// <summary>
+        ///     Use the default Cache Behavior of the client.
+        /// </summary>
+        /// <seealso cref="DiscordClientConfig.DefaultStateBehavior"/>
+        Default = 0,
+        /// <summary>
+        ///     The entity will only be retrieved via a synchronous cache lookup.
+        ///
+        ///     For the default <see cref="IStateProvider"/>, this is equivalent to using <see cref="CacheOnly"/>
+        /// </summary>
+        /// <remarks>
+        ///     This flag is used to indicate that the retrieval of this entity should not leave the
+        ///     synchronous path of the <see cref="System.Threading.Tasks.ValueTask"/>. When true,
+        ///     the calling method *should* not ever leave the calling task, and never generate an async
+        ///     state machine.
+        ///
+        ///     Bear in mind that the true behavior of this flag depends entirely on the <see cref="IStateProvider"/> to
+        ///     abide by design implications of this flag. Once Discord.Net has called out to the state provider with this
+        ///     flag, it is out of our control whether or not an async method is evaluated.
+        /// </remarks>
+        SyncOnly = 1,
```

Comment:
Possible need for clarification: What's the difference between `SyncOnly` and `CacheOnly`?

Reply:
That's an excellent thought. You should probably just take that last paragraph and inject it into the XML.